android/core/src/main/java/ai/djl/android/core/BitmapImageFactory.java

```diff
@@ -86,24 +86,29 @@ public Image fromImage(Object image) {
     @Override
     public Image fromNDArray(NDArray array) {
         Shape shape = array.getShape();
+        int height = (int) shape.get(1);
+        int width = (int) shape.get(2);
         if (shape.dimension() != 3) {
             throw new IllegalArgumentException("Shape should only have three dimension follow CHW");
         }
         if (shape.get(0) == 1) {
             throw new UnsupportedOperationException("Grayscale image is not supported");
+        } else if (shape.get(0) != 3) {
+            if (shape.get(2) == 3) {
```

Comment:
Based on the error message, shouldn't the case of `shape.get(2) == 1` also be handled?

Reply:
Added a check for this